instrumentation/github.com/aws/aws-lambda-go/otellambda/wrapHandler.go

```diff
@@ -38,6 +40,10 @@ func (h wrappedHandler) Invoke(ctx context.Context, payload []byte) ([]byte, err
 
 	response, err := h.handler.Invoke(ctx, payload)
 	if err != nil {
+		if h.instrumentor.configuration.RecordError {
+			span.RecordError(err, trace.WithStackTrace(h.instrumentor.configuration.RecordStackTrace))
+		}
+		span.SetStatus(codes.Error, err.Error())
```

Comment:
I'm not sure this is correct.  There are situations where an error may be returned from the handler but that should not result in the span status being set to error.  For instance, if the handler is supporting an HTTP-based service and an error would indicate a `4xx` status code, the span status [MUST](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#status) be left unset.  Similarly, the status message shouldn't be set when it can be inferred from other data such as the HTTP status code.

Given that the instrumented handler has access to this span and can set the status as appropriate I would omit setting it here.